AsyncDisplayKit/ASMapNode.mm

```diff
@@ -0,0 +1,236 @@
+/* Copyright (c) 2014-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+#import "ASMapNode.h"
+#import <AsyncDisplayKit/ASInsetLayoutSpec.h>
+#import <AsyncDisplayKit/ASCenterLayoutSpec.h>
+#import <AsyncDisplayKit/ASThread.h>
+
+@interface ASMapNode()
+{
+  ASDN::RecursiveMutex _propertyLock;
+  CGSize _nodeSize;
+  MKMapSnapshotter *_snapshotter;
+  MKMapSnapshotOptions *_options;
+  CGSize _maxSize;
+  NSArray *_annotations;
+}
+@end
+
+@implementation ASMapNode
+
+@synthesize liveMap = _liveMap;
+@synthesize mapSize = _mapSize;
+@synthesize automaticallyReloadsMapImageOnOrientationChange = _automaticallyReloadsMapImageOnOrientationChange;
+@synthesize mapDelegate = _mapDelegate;
+
+- (instancetype)initWithCoordinate:(CLLocationCoordinate2D)coordinate
+{
+  if (!(self = [super init])) {
+    return nil;
+  }
+  self.backgroundColor = ASDisplayNodeDefaultPlaceholderColor();
+  _automaticallyReloadsMapImageOnOrientationChange = YES;
+  _liveMap = NO;
+
+  _options = [[MKMapSnapshotOptions alloc] init];
+  _options.region = MKCoordinateRegionMakeWithDistance(coordinate, 1000, 1000);;
+  _mapImage = [[ASImageNode alloc]init];
+  _mapImage.clipsToBounds = YES;
+  [self addSubnode:_mapImage];
+  _maxSize = self.bounds.size;
+  return self;
+}
+
+- (void)addAnnotations:(NSArray *)annotations
```

Comment:
I agree setAnnotations: makes more conceptual sense. This is basically what the method already does, since it just takes the new annotations and redraws. So more of a naming change. Possibly will support passing `nil` or an empty array to clear annotations, does ASDK have any guidelines on this?